lonestar/analytics/cpu/k-shortest-paths/ksssp_cli.cpp

```diff
@@ -0,0 +1,178 @@
+    "threadSpin",
+    cll::desc("If enabled, threads busy-wait for work rather than use "
+              "condition variable (default false)"),
+    cll::init(false));
+
+std::string
+AlgorithmName(SsspPlan::Algorithm algorithm) {
+  switch (algorithm) {
+  case SsspPlan::kDeltaTile:
+    return "DeltaTile";
+  case SsspPlan::kDeltaStep:
+    return "DeltaStep";
+  case SsspPlan::kDeltaStepBarrier:
+    return "DeltaStepBarrier";
+  default:
+    return "Unknown";
+  }
+}
+
+template <typename Weight>
+static void
+OutputResults(
+    katana::PropertyGraph* pg, std::string node_distance_prop,
+    std::string output_filename = "output") {
+  auto r = pg->GetNodePropertyTyped<Weight>(node_distance_prop);
+  if (!r) {
+    KATANA_LOG_FATAL("Error getting results: {}", r.error());
+  }
+  auto results = r.value();
+  KATANA_LOG_DEBUG_ASSERT(
+      uint64_t(results->length()) == pg->topology().NumNodes());
+
+  writeOutput(
+      outputLocation, results->raw_values(), results->length(),
+      output_filename);
+}
+}  // namespace
+
+int
+main(int argc, char** argv) {
+  std::unique_ptr<katana::SharedMemSys> G =
+      LonestarStart(argc, argv, name, desc, url, &inputFile);
+
+  katana::StatTimer totalTime("TimerTotal");
+  totalTime.start();
+
+  if (symmetricGraph) {
+    KATANA_LOG_WARN(
+        "This application requires a symmetric graph input;"
+        " Using the -symmetricGraph flag "
+        " indicates that the input is a symmetric graph and can be used as it "
+        "is.");
+  }
+
+  katana::gInfo("Reading from file: ", inputFile, "\n");
+  std::unique_ptr<katana::PropertyGraph> pg =
+      MakeFileGraph(inputFile, edge_property_name);
+
+  std::cout << "Read " << pg->topology().NumNodes() << " nodes, "
+            << pg->topology().NumEdges() << " edges\n";
+
+  if (algo == SsspPlan::kDeltaStep || algo == SsspPlan::kDeltaTile) {
+    katana::gInfo("Using delta-step of ", (1 << stepShift), "\n");
+    KATANA_LOG_WARN(
+        "Performance varies considerably due to delta parameter.\n");
+    KATANA_LOG_WARN("Do not expect the default to be good for your graph.\n");
+  }
+
+  std::cout << "Running " << AlgorithmName(algo) << " algorithm\n";
+
+  SsspPlan plan;
+  switch (algo) {
+  case SsspPlan::kDeltaTile:
+    plan = SsspPlan::DeltaTile(stepShift);
+    break;
+  case SsspPlan::kDeltaStep:
+    plan = SsspPlan::DeltaStep(stepShift);
+    break;
+  case SsspPlan::kDeltaStepBarrier:
+    plan = SsspPlan::DeltaStepBarrier(stepShift);
+    break;
+  default:
+    KATANA_LOG_FATAL("Invalid algorithm selected");
+  }
+
+  if (startNode >= pg->topology().size() ||
+      reportNode >= pg->topology().size()) {
+    KATANA_LOG_ERROR(
+        "failed to set report: ", reportNode,
+        " or failed to set source: ", startNode, "\n");
+    assert(0);
+    abort();
+  }
+
+  katana::TxnContext txn_ctx;
+
+  auto pg_result = Ksssp(
+      pg.get(), startNode, reportNode, &txn_ctx, algoReachability, numPaths,
```

Comment:
I thought you were supposed to pass `edge_property_name` so that edge weights can be loaded